.github/workflows/os-zoo.yml

```diff
@@ -37,7 +38,7 @@ jobs:
         ref: ${{ matrix.branch }}
     - name: config
       run: |
-        ./config --banner=Configured no-shared -Wall -Werror enable-fips --strict-warnings -DOPENSSL_USE_IPV6=0 \
+        ./config --banner=Configured no-shared -Wall -Werror enable-fips --strict-warnings \
```

Comment:
So I see no need for this change given #24748

Reply:
Hmm, also the `-Wno-switch-default` change would not be necessary
if the `-Wswitch-default` was removed in the Configure script.

Reply:
@kroeckx I think the removal of OPENSSL_USE_IPV6 is valid here as it seems it was added because ipv6 wasn't available on the alpine container when the job was written, and that is no longer the case

Reply:
@bernd-edlinger I think the larger question is "do we want to disable default case checking".  We're disabling it here, because its breaking our CI, and thats the priority at the moment.  But it seems like a legitimate warning to me, for which the proper fix is to actually add a default case to all our switch statements.  Removing it from Configure says to me that we're ok with that warning globally, which I'm not sure is the case

Reply:
@nhorman We absolutely must drop this warning in Configure because otherwise we cannot use --strict-warnings at all with new clang. This is something to do on all active branches.

Whether we want to clean the code with this warning enabled is something for OTC to decide. If we decide to do so, it will be a subsequent PR for the master branch that will need some effort to do to clean the code. It should NOT be applied to stable branches as we do not apply such kind of refactoring there.

Reply:
This warning was dead all the time, and once that is fixed in the clang compiler,
this turns out to be annoying, and unfortunately something that nobody payed attention to.

I have no gut feeling how many switch statements this would affect.
I would suggest to disable the warning in all branches for now, and evaluate later,
how much changes it causes but it would probably a lot of changes, and rewriting
lots of code will therefore be something for master branch only.

Reply:
@t8m every other clang version we use, as @bernd-edlinger notes is clang-16 or -17 (for now) so we have time on this before things get upgraded, as older clang versions ignore the detection of this warning..

@bernd-edlinger the number of switch statements we need to address is easily in the dozens, perhaps more than 100.

I think removing the check from the Configure script now is the wrong move.  We need it here for alpine specifically, but removing it globally invites us to minimize this problem and forget about it.  I think we should leave it enabled and use it as a game clock to fix the issue properly before other environments start upgrading to newer clang versions.

I've opened https://github.com/openssl/openssl/issues/24751 to track that, and put an otc hold on it for discussion